cmds/exp/traceopen/traceopen_linux.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2012-2024 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build !tinygo && linux && (amd64 || riscv64 || arm64)
+// +build !tinygo
+// +build linux
+// +build amd64 riscv64 arm64
+
+// traceopen traces a process trees file opens.
+// It is intended to make building small containers easier, so that,
+// e.g., gigantic containers (e.g. the 25G Nemo ML container)
+// can be cut down.
+//
+// Synopsis:
+//
+//	traceopen <command> [args...]
+//
+// Description:
+//
+//	trace all file opens for a tree of processes.
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/strace"
+	"golang.org/x/sys/unix"
+)
+
+var errUsage = errors.New("usage: traceopen [-o <outputfile>] <command> [args...]")
+
+type params struct {
+	output string
+}
+
+// PrintTraces prints only trace entry events, and only
+// for a subset of system calls, and only the file name
+// and open mode.
+func PrintFiles(w io.Writer) strace.EventCallback {
+	return func(t strace.Task, record *strace.TraceRecord) error {
+		if record.Event != strace.SyscallEnter {
+			return nil
+		}
+		var n strace.Addr
+		var mode uint
+		var cwd string
+		var err error
+
+		e := record.Syscall
+		switch e.Sysno {
+		case syscall.SYS_OPEN:
```

Comment:
This will not work on arm64, so maybe remove arm64 from build tags. We can address this later.